autopxd/__init__.py

```diff
@@ -303,16 +303,15 @@ def lines(self):
 
 
 def preprocess(code, extra_cpp_args=[]):
-    proc = subprocess.Popen([
-        'cpp', '-nostdinc', '-D__attribute__(x)=', '-I', BUILTIN_HEADERS_DIR,
-    ] + extra_cpp_args + ['-'], stdin=subprocess.PIPE, stdout=subprocess.PIPE)
-    result = []
-    result.append(proc.communicate(input=code.encode('utf-8'))[0])
-    while proc.poll() is None:
-        result.append(proc.communicate()[0])
+    proc = subprocess.Popen(
+        ['cpp', '-nostdinc', '-D__attribute__(x)=',
+            '-I', BUILTIN_HEADERS_DIR] +
+        extra_cpp_args + ['-'],
+        stdin=subprocess.PIPE, stdout=subprocess.PIPE, universal_newlines=True)
+    result = proc.communicate(input=code)[0]
```

Comment:
communicate should be called only once. doc:

> Read data from stdout and stderr, until end-of-file is reached. Wait for process to terminate.

Use `universal_newlines` to get platform independent source code strings. Note doc:

> The type of input must be bytes or, if universal_newlines was True, a string.
> communicate() returns a tuple (stdout_data, stderr_data). The data will be bytes or, if universal_newlines was True, strings.